src/debugger.rs

```diff
@@ -183,36 +218,91 @@ impl ContextDebugExt for Context {
 
 #[cfg(test)]
 mod tests {
-    use assert_cmd::Command;
+    use crate::{Context, ContextPeopleExt};
     use predicates::str::contains;
+    use std::{cell::RefCell, io::Write, rc::Rc};
+
+    use super::build_repl;
+
+    #[derive(Clone)]
+    struct StdoutMock {
+        storage: Rc<RefCell<Vec<u8>>>,
```

Comment:
Why does this have to be an Rc?

Reply:
I did this to get around having to pass in a reference that can be stored in the cli struct and modify the vec, but also be able to check it later for tests once you release the cli's reference